src/sentry/rules/actions/utils.py

```diff
@@ -36,12 +36,36 @@ def generate_diff_labels(
     rule: Rule,
     changed_data: DefaultDict[str, list[str]],
     key: str,
-    statement: str,
 ) -> DefaultDict[str, list[str]]:
+    added_statement = "Added {} '{}'"
+    removed_statement = "Removed {} '{}'"
+    added_items = []
+
+    # Added items that are not seen in the prior rule data
     for data in prior_state.get(key, []):
         if data not in present_state.get(key, []):
             label = generate_rule_label(rule.project, rule, data)
-            changed_data[data["id"]].append(statement.format(label))
+            changed_data[data["id"]].append(added_statement.format(key[:-1], label))
+            added_items.append(data["id"])
+
+    # Check if the id is in both but the data has changed
+    for item in added_items:
+        if changed_data.get(item):
+            for data in prior_state.get(key, []):
+                for present_data in present_state.get(key, []):
```

Comment:
ditto with present state

Reply:
Also - we're doing a double for loop to grab the item in the present/prior sets

is it possible for there to be a prior_data that matches multiple present_data's? (and vice versa)

Reply:
That would mean you edited an action condition and then added another that's exactly the same, right?

Reply:
Not entirely sure but sounds like the answer is yes? so also yes to vice versa? 😅 

So in that case the double forloop may be fine,
but regarding my other comments, sounds like we could make this a lot easier by pre-compiling the data beforehand, pulling together some easier to iterate through datastruct, and then distilling changes?

Reply:
🤔 I thought I already addressed your suggestions, I guess I don't know what you mean then.